### DISCUSSION TITLE
Shared Element Transitions

### DISCUSSION THREAD

**Discussion by kennetpostigo** [expo/router#150] [Q&A]
Are there any examples of how to do shared element transitions with expo-router?

**Comment by EvanBacon** [comment]
No, we use `react-native-screens` for stack navigation, this does not currently have SET but it is in the works. https://github.com/software-mansion/react-native-screens/pull/1591 once landed, we can implement fairly easily in Expo Router.

**Reply by alanschwarz** [reply]
any news on SET?

**Comment by ThatGuySam** [comment]
Looks like it's way simpler than the existing Expo Shared Element example. 

You just use a `Animated.*` component from reanimated and add a `sharedTransitionTag="someMatchingTag"`

```tsx
import Animated from 'react-native-reanimated'

function Screen1({ navigation }) {
  return (
    <View style={{ flex: 1 }}>
      <Animated.View
        style={{ width: 150, height: 150, backgroundColor: 'green' }}
        sharedTransitionTag="sharedTag"
      />
      <Button title="Screen2" onPress={() => router.push('/screen2')} />
    </View>
  );
}

function Screen2({ navigation }) {
  return (
    <View style={{ flex: 1, marginTop: 50 }}>
      <Animated.View
        style={{ width: 100, height: 100, backgroundColor: 'green' }}
        sharedTransitionTag="sharedTag"
      />
      <Button title="Screen1" onPress={() => router.push('/screen1')} />
    </View>
  );
}
```

**Reply by jzrbsv** [reply]
@ThatGuySam thanks for sharing. I'm gonna try this

**Comment by eyoelmeles** [comment]
@ThatGuySam 
I have tried to duplicate you example, and it doesn't work. The following is my code. did i miss something?
```ts
import Animated from "react-native-reanimated";
import { router } from "expo-router";

const { width } = Dimensions.get("window");

const ScreenA = () => {
  return (
    <View style={styles.container}>
      <Animated.View
        style={styles.sharedContainerA}
        sharedTransitionTag="tagger"
      >
        <Text style={styles.textStyleA}>Hello world</Text>
      </Animated.View>
      <TouchableOpacity
        onPress={() => {
          router.push("/shared_element/shared_component_b");
        }}
      >
        <Text>Go to B</Text>
      </TouchableOpacity>
    </View>
  );
};

export default ScreenA;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  sharedContainerA: {
    width: width * 0.6,
    height: 400,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "tomato",
  },
  textStyleA: {
    fontSize: 18,
    fontWeight: "300",
  },
});

// Screen B

import Animated from "react-native-reanimated";

const { width } = Dimensions.get("window");

const ScreenB = () => {
  return (
    <View style={styles.container}>
      <Animated.View
        style={styles.sharedContainerB}
        sharedTransitionTag="tagger"
      >
        <Text style={styles.textStyleB}>Hello world</Text>
      </Animated.View>
      <Link
        href={{
          pathname: "/shared_element",
        }}
      >
        Back
      </Link>
      <Text>SharedComponents</Text>
    </View>
  );
};

export default ScreenB;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "flex-start",
    alignItems: "center",
  },

  sharedContainerB: {
    width: width * 0.9,
    height: 600,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "tomato",
  },
  textStyleB: {
    fontSize: 22,
    fontWeight: "300",
  },
});
```ts

**Reply by 0rtbo** [reply]
yeah ive tried it too. i got it to work but its quite buggy

**Reply by ThatGuySam** [reply]
It might be an issue with the newest Expo/Expo Router version. 

I ended up not using it because the start or end had a visual issue that's was too tricky to work around for what I was doing.

**Comment by lewisdonovan** [comment]
Any update on this? Kinda nuts that we can't do this in Expo, it's such a standard feature of apps nowadays.

**Comment by QWin13PT** [comment]
Any updates? Looking forward for this functionality

**Comment by ThatGuySam** [comment]
Yeah, it's not working for me either anymore. 

I tried it here with Expo 51 for anyone who wants to take a crack at it: 
https://github.com/ThatGuySam/expo-router-transition-demo

I think the issue might be that it needs to be the same "Stack" or "Navigator"

Here's the docs for doing it with React Navigation, which Expo Router uses under the hood: 
https://reactnavigation.org/docs/7.x/shared-element-transitions?config=dynamic

There's some disconnect between the expected React Navigation implementation and Expo Router.

**Reply by themtaysw** [reply]
as of today, expo-router uses v6 of react-navigation not v7... they are about to release expo [SDK52](https://expo.dev/changelog/2024/10-24-sdk-52-beta) which should be using v7 so probably best to do now ismjust wait a few weeks and hopefully have it then 😄

**Comment by sniperyork** [comment]
Any updates on this? hopefully this feature is coming soon.

**Comment by i8ramin** [comment]
Also very interested in this feature. Subscribing for details. thx

**Comment by adamblvck** [comment]
Yup, shared transitions don't seem to work when one sits in `tabs`, and the other in a `stack`.

**Reply by joekendal** [reply]
I have solved it by using a parent stack and tab child + stack child. See explanation: https://github.com/expo/expo/issues/27606#issuecomment-2464672425

**Comment by buildnewapp** [comment]
I have tried many methods, but the expo router does not support shared elements. Time: November 8, 2024 21:11:14

**Comment by joekendal** [comment]
@buildnewapp well I have it working with `expo@52.0.0-preview.22` and `expo-router@4.0.0-preview.14` no problem. The issue is probably your use of shared elements. Can you elaborate how you are trying to use shared elements?

I am using <Animated.View sharedTransitionTag={id}> from `react-native-reanimated@3.16.1` and its working as explained [here](https://github.com/expo/expo/issues/27606#issuecomment-2464672425)

**Reply by buildnewapp** [reply]
I don't want to upgrade to the beta version yet. Let's wait for the shared element
I mainly use pictures and video lists for detailed scenarios

**Comment by joekendal** [comment]
It doesn't work with the New Architecture though and the maintainer of shared elements has quit. So we have to make sure:

app.json
```
...
"newArchEnabled": false
...
```

is set going forward.

**Reply by joekendal** [reply]
the maintainer who quit was hired at expo but does not look like expo are maintaining it

**Comment by Th1nhNg0** [comment]
I hope we can fix this soon

**Comment by ElJosecito** [comment]
There is any update in this issue?

I really want to make this behavior in my app and i'm using expo 52 realesed version, with the file based routing.

**Comment by marcopolodesign** [comment]
Any updates here? Using SharedTransitionTag but getting the default navigation of my Stack - do I need to overwrite this somehow?

**Comment by guillaumehcht** [comment]
Any news? Trying to implement the same thing...

**Comment by RogierZeebregts** [comment]
Yeah, waiting for an update too

**Comment by AliMousa27** [comment]
Any updates? I'm anticipating this feature!

**Comment by LegendProjects** [comment]
?

**Comment by tofful** [comment]
Some good news would be much appreciated please. This feature will add tones of value to expo packages

**Comment by pixelmaxxing** [comment]
++ I am waiting for this too please would be an amazing feature and it's not an irrelevant one. Most of the apps have shared transitions.

**Comment by henrique-c-ladeira** [comment]
Tried to implement the same thing and ended up here! Hope to have this feature soon as well.

**Comment by victorcolina22** [comment]
I have the same issue trying to implement it with expo 52 D:

**Comment by MirzaFreddy** [comment]
come on guys do something XD

**Comment by SuzuyaBish** [comment]
Please guys, 2+ years now

**Comment by manyajain27** [comment]
its a shame that you can't really do this properly in expo. its such a standard feature!!

**Comment by irvanahmadp** [comment]
Hi guys, any update in this issue?
